packages/nimble-components/src/table/specs/table-row-selection.md

```diff
@@ -0,0 +1,136 @@
+-   [enableSubRowSelection](https://tanstack.com/table/v8/docs/api/features/row-selection#enablesubrowselection) - `true`
+
+As mentioned above, TanStack does not update the selection state when the data changes. Therefore, it will be the nimble table's responsiblity to ensure that the set of selected records only includes records that currently exist in the data when `setData()` is called and to clear the selection when `id-field-name` changes.
+
+### Validation
+
+The table will be invalid when a selection mode other than `TableRowSelectionMode.none` is set and an `id-field-name` is not set. A new `idFieldNameNotConfigured` key will be added to the `TableValidity` object to represent this invalid configuration.
+
+### Accessibility
+
+Accessbility guidance taken from [MDN role="row"](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Roles/row_role)
+
+-   Rows that are currently selected will have `aria-selected="true"` configured.
+-   Rows that are currently not selected but can be selected will have `aria-selected="false"` configured.
+-   When selection is not enabled in the table, `aria-selected` will not be set on any rows.
+
+Note: The guidance above is when rows are within a component with a role of `grid` or `treegrid`. However, the nimble table currently has a role of `table`.
+
+## Alternative Implementations / Designs
+
+### Modify TanStack configuration approach
+
+Rather than specifying grouped rows as not selectable, we could have the TanStack options specify that all rows (including grouped rows) support selection. The downside of this approach is that the selection state within TanStack will include row IDs for grouped rows. Therefore, we would need to filter out IDs associated with grouped rows prior to returning the selection state to a client of the table.
+
+### Allow selection on tables without an `id-field-name` set
+
+Instead of having validation preventing enabling selection in the table without an `id-field-name`, the table could support selection while using generated IDs from TanStack. However, this would make deleting the selection state for removed rows more difficult, and it is also unlikely to enable easier use for a client because the record IDs associated with selection don't provide meaning to a client without `id-field-name` set.
+
+## Open Issues
+
+-   There is [an issue in TanStack](https://github.com/TanStack/table/issues/4759) with deselecting a grouped row with the proposed configuration. There is also [an open PR](https://github.com/TanStack/table/pull/4760) with a fix. If the issue isn't resolved within TanStack, we will need to put a workaround in the nimble-table.
+-   The `aria-selected` attribute on rows is intended to be used for role `grid` or `treegrid`. However, the nimble table currently has a role of `table`. Should the role of the table change to `grid` when selection is enabled? Should the table always have a role of `grid`?
```

Comment:
The sense I got from the keyboard interaction IxD spec was that the table would always behave with role `grid`. Unless we can think of reasons not to, I'd be ok with making that change now even though we haven't done the rest of the work to make it behave like a grid yet.

Reply:
I think you're right. I'll remove this open issue and just note in the accessibility section that we will change the role of the table to `grid`.